File: Baekjoon_Algorithm/Baek_Algorithm_python/시간복잡도/2517_달리기.py





if __name__ == "__main__" :
    N = int(input())
    runner = [0 for _ in range(N)]
    result = []
    for i in range(N) :
        runner[i] = int(input())
    for i in range(N) :
        current_runner = runner[i]
        count = 0
        for j in range(i) :
            if current_runner > runner[j] :
                count += 1
        num = i + 1 - count
        result.append(num)



    for i in range(N) :
        print(result[i])

File: Baekjoon_Algorithm/Algorithm_cpp/시간복잡도/2517_달리기.cpp
#include <iostream>
using namespace std;
long N ;
long long runner[500001] ;
long long result[500001] ;
long long current_runner = 0 ;
long long idx = 0 ;
long long num = 0 ;

int main() {
    cin >> N ;
    
    for(int i = 0 ; i < N ; i++)
        cin >> runner[i] ;
    for(int i = 0 ; i < N ; i++){
        idx = 0 ; 
        current_runner = runner[i] ;
        for(int j = 0 ; j < i ; j++){
            if (current_runner > runner[j])
                idx += 1 ;
        }
        num = i + 1 - idx ;
        result[i] = num ;
    }
    
    for(int i = 0 ; i < N ; i++)
        cout << result[i] << endl ;
    
}

